File: Models/BookModel.cs
namespace Books.Models
{
    public class BookModel
    {

        public int Id { get; set; }
        public string Title { get; set; }
        public int TotalPages { get; set; }
        public PriceModel Price { get; set; }
        public bool IsPublished { get; set; }
        public DateTime PublishedOn { get; set; }
        public List<AuthorModel>  Authors { get; set; }
    }
}

File: src/app/books/models/book.model.ts
import { PriceModel } from '../models/price.model';


export class BookModel {
    public id = 'number';
    public title = 'string';
    public totalPages = 'number';
    public author = 'string';
    public price = PriceModel;
    public isPublished = 'boolean';
    public publishedOn = Date;
    public authors = 'string[]';
    
    }